### PR TITLE
1581 catch fact error concerning inconsistent ontology

### LINKED ISSUE #1581: catch Fact error concerning inconsistent ontology
I want to prove anything in an inconsistent ontology:

```
logic OWL
ontology Inconsistency =
   Class: owl:Thing SubClassOf: 
                owl:Nothing 
   Class: A
   Class: B SubClassOf:  Annotations: Implied "true"^^xsd:string
                A

end
```

But then I get

```
Internal error.
FactProver: de.unibremen.informatik.FactProver.FactProver@7c2d0928

PhysicalIRI: file:///tmp/implied_Inconsistency_Ax211846.owl

GoalIRI: file:///tmp/implied_Inconsistency_Ax211846.owl.entail.owl

Inconsistent ontology
FaCT++.Kernel: Reasoner for the SROIQ(D) Description Logic, 64-bit
Copyright (C) Dmitry Tsarkov, 2002-2014. Version 1.6.3 (30 May 2014)
```

Instead, I would expect that the proof succeeds.

### PR DESCRIPTION
**Pull request by b-gehrke**
Closes #1581.

Errors thrown by provers for inconsistent ontologies are caught and reinterpreted as successful prove.

### CODE REVIEW

File: OWL2/ProveFact.hs [hunk 1]
```diff
@@ -207,14 +207,12 @@ runFact sps cfg saveFact thName nGoal = do
         writeFile tmpFileName prob
         writeFile (tmpFileName ++ ".entail.owl") entail
       mExit <- runTimedFact tmpFileName prob (Just entail) tLimit
-      ((err, retval), output, tUsed) <- case mExit of
-            Just (b, ex, output, t_u) -> if b then do
```
Comment (mcodescu) [inline OWL2/ProveFact.hs:211]: Fix warning about shadowing.

**Review by mcodescu — COMMENTED** [review]
Things work as expected with Pellet. With Fact I still get:
```
Internal error.
FactProver: de.unibremen.informatik.FactProver.FactProver@72c28d64

PhysicalIRI: file:///tmp/./inconsist_Inconsistency_Ax28491-0.owl

GoalIRI: file:///tmp/./inconsist_Inconsistency_Ax28491-0.owl.entail.owl

Inconsistent ontology
FaCT++.Kernel: Reasoner for the SROIQ(D) Description Logic, 64-bit
Copyright (C) Dmitry Tsarkov, 2002-2014. Version 1.6.3 (30 May 2014)
```
Do I need a newer version of the prover?

**Comment by b-gehrke** [comment]
> Things work as expected with Pellet. With Fact I still get:
> [...]
> 
> Do I need a newer version of the prover?

I think you just need to rebuild the OWL2 jar files. Either with `make jars` or `cd OWL2/java && ant`.

File: OWL2/ProveFact.hs (hunk 1 above)
Comment (b-gehrke) [reply OWL2/ProveFact.hs:211]: Fixed

**Comment by mcodescu** [comment]
When I call either of the two commands you mentioned, I get `BUILD FAILED
/home/mcodescu/repos/Hets2/OWL2/java/build.xml:75: Unable to delete file /home/mcodescu/repos/Hets2/OWL2/java/build/OWL2Parser.jar`. Is it normal to have the build folder owned by root?

**Comment by mcodescu** [comment]
After deleting the content of OWL2/java/build and running `sudo make jars`, I still get an inconsistency error for Fact.

**Comment by b-gehrke** [comment]
I tried it on a clean repository and it gets proven with Fact. I used the following commands:

```bash
# Cleaning
git checkout -- .
git clean -xdf

# Building
make hets
make jars

# Executing
./hets -g reason.dol
```

**Review by mcodescu — APPROVED** [review]
:+1:

**Pull request merged**